src/test/java/org/openrewrite/java/spring/data/MigrateAuditorAwareToOptionalTest.java

```diff
@@ -0,0 +1,362 @@
+              
+              import java.util.Objects;
+              
+              public class Configuration {
+                  public AuditorAware<String> auditorAware() {
+                      return () -> {
+                          return Objects.toString("admin");
+                      };
+                  }
+              }
+              """, """
+              package sample;
+              
+              import org.springframework.data.domain.AuditorAware;
+              
+              import java.util.Objects;
+              import java.util.Optional;
+              
+              public class Configuration {
+                  public AuditorAware<String> auditorAware() {
+                      return () -> {
+                          return Optional.ofNullable(Objects.toString("admin"));
+                      };
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void rewriteInterfaceInstantiation() {
+        //TODO Question for TIM: how to get rid of the types? I have the imports.
+        //- public Optional<String> getCurrentAuditor() {
+        //+ public java.util.Optional<java.lang.String> getCurrentAuditor() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              package sample;
+              
+              import org.springframework.data.domain.AuditorAware;
+              
+              public class Configuration {
+                  public AuditorAware<String> auditorAware() {
+                      return new AuditorAware<String>() {
+                          @Override
+                          public String getCurrentAuditor() {
+                              return "admin";
+                          }
+                      };
+                  }
+              }
+              """, """
+              package sample;
+              
+              import org.springframework.data.domain.AuditorAware;
+              
+              import java.util.Optional;
+              
+              public class Configuration {
+                  public AuditorAware<String> auditorAware() {
+                      return new AuditorAware<String>() {
+                          @Override
+                          public Optional<java.lang.String> getCurrentAuditor() {
+                              return Optional.ofNullable("admin");
+                          }
+                      };
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void rewriteMethodReference() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              package sample;
+              
+              import org.springframework.data.domain.AuditorAware;
+              
+              public class Configuration {
+                  public AuditorAware<String> auditorAware() {
+                      return this::getCurrentAuditor;
```

Comment:
Optionally there's this utility that for now was tucked away; might make this easier if we leverage that util.
https://github.com/openrewrite/rewrite-spring/blob/298ba3aeffc5cf1035a0e27c85caf5ed2f3ec0b5/src/main/java/org/openrewrite/java/spring/util/concurrent/MemberReferenceToMethodInvocation.java#L31

Reply:
If that helps here then we can make it public and pull it up; I'd intentionally written it as a reusable component, that I hadn't proactively pulled up just yet until there was another use case for it.

Reply:
should we pull it up to rewrite-java?

Reply:
I'd say not yet. Both uses are in rewrite-spring for now